### ISSUE TITLE
Some lua tests fail with lua 5.4

### ISSUE THREAD

**Issue by vp1981** [vim/vim#8042]
**Describe the bug**
Some lua tests:
```
Failures:                                                                                                                                                                                                                                       
        From test_lua.vim:                                                                                                                                                                                                                      
        Found errors in Test_lua_blob():                                                                                                                                                                                                        
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[473]..function RunTheTest[44]..Test_lua_blob line 29: Expected '[string "luaeval"]:1: attempt to call a nil value (method ''xyz'')' but got '[string "luaev
al"]:1: method ''xyz'' is not callable (a nil value)': let x = luaeval("lb:xyz(3)")                                                                                                                                                             
        Found errors in Test_lua_buffer_insert():                                                                                                                                                                                               
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[473]..function RunTheTest[44]..Test_lua_buffer_insert line 11: Expected '[string "vim chunk"]:1: attempt to call a nil value (method ''xyz'')' but got '[st
ring "vim chunk"]:1: method ''xyz'' is not callable (a nil value)': lua vim.buffer():xyz()                                                                                                                                                      
        Found errors in Test_lua_list():                                                                                                                                                                                                        
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[473]..function RunTheTest[44]..Test_lua_list line 34: Expected '[string "luaeval"]:1: attempt to call a nil value (method ''xyz'')' but got '[string "luaev
al"]:1: method ''xyz'' is not callable (a nil value)': let x = luaeval("ll:xyz(3)")                                                                                                                                                             
        Found errors in Test_lua_luado():                                                                                                                                                                                                       
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[473]..function RunTheTest[44]..Test_lua_luado line 10: Expected '[string "vim chunk"]:1: attempt to call a nil value (global ''func'')' but got '[string "v
im chunk"]:1: global ''func'' is not callable (a nil value)': luado func()                                                                                                                                                                      
        Found errors in Test_lua_window():                                                                                                                                                                                                      
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[473]..function RunTheTest[44]..Test_lua_window line 12: Expected '[string "luaeval"]:1: attempt to call a nil value (field ''xyz'')' but got '[string "luae
val"]:1: field ''xyz'' is not callable (a nil value)': let n = luaeval('vim.window().xyz()') 
```
failed to run with lua 5.4. Seems that lua 5.4 has changed error message. Should I bother about failding tests with lua 5.4 or compile vim vs lua 5.3 only?

**Additional context**
OS: Archlinux x86_64
GCC: gcc (GCC) 10.2.0
LUA: lua 5.4.3

**Comment by brammool** [comment]
> failed to run with lua 5.4. Seems that lua 5.4 has changed error
> message. Should I bother about failding tests with lua 5.4 or compile
> vim vs lua 5.3 only?

We already adjust the expected errors for Lua 5.3:

 call assert_fails('luado func()',
        \ s:lua_53_or_later
        \ ? "[string \"vim chunk\"]:1: attempt to call a nil value (global 'func')"
        \ : "[string \"vim chunk\"]:1: attempt to call global 'func' (a nil value)")

Apparently we need to add a s:lua_54_or_later check.

-- 
On the other hand, you have different fingers.
                                      -- Steven Wright

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by chrisbra** [comment]
@vp1981 could you prepare a PR to fix those failures?

**Comment by vp1981** [comment]
@chrisbra, I'll try to make one on weekend.

**Comment by vim-ml** [comment]
Hi,

I have made the changes. I will create a PR for this.

- Yegappan

**Comment by lacygoill** [comment]
Possibly fixed by [8.2.2696](https://github.com/vim/vim/releases/tag/v8.2.2696) and [8.2.2699](https://github.com/vim/vim/releases/tag/v8.2.2699).

**Comment by vp1981** [comment]
Hi.
vim 8.2.2706 compiles fine and passes all tests.
